go/vt/vttablet/tabletserver/query_engine.go

```diff
@@ -300,6 +295,13 @@ func (qe *QueryEngine) GetPlan(ctx context.Context, logStats *tabletenv.LogStats
 	qe.mu.RLock()
 	defer qe.mu.RUnlock()
 	statement, err := sqlparser.Parse(sql)
+	if !skipQueryPlanCache && !sqlparser.SkipQueryPlanCacheDirective(statement) {
```

Comment:
Moving this here means that even when the plan is in the cache and can be re-used, we still parse the sql (because we want to check the Skip directive).
Is this worth it? 
Is it something that should be benchmarked?

Reply:
I think it is a good idea to add a micro-benchmark on top of `GetPlan`, if the results are not conclusive we can always add the `Benchmark Me` label to this PR.